src/Access/UsersConfigAccessStorage.cpp

```diff
@@ -487,7 +489,13 @@ namespace
 
             if (key == "constraints" || key.starts_with("constraints["))
             {
-                profile->elements.merge(parseSettingsConstraints(config, profile_config + "." + key, access_control));
+                profile->elements.merge(parseSettingsConstraints(config, profile_config + "." + key, access_control, SettingsProfileElement::RangeKind::Constrain));
+                continue;
+            }
+
+            if (key == "allow" || key.starts_with("allow["))
+            {
+                profile->elements.merge(parseSettingsConstraints(config, profile_config + "." + key, access_control, SettingsProfileElement::RangeKind::Allow));
                 continue;
```

Comment:
It seems it will be a bit difficult to undestand, for example:

```
<clickhouse>
    <profiles>
        <profile1>
            <max_memory_usage>10000000000</max_memory_usage>
            <readonly>1</readonly>
            <constraints>
                <max_memory_usage>
                    <min>5000000000</min>
                    <max>20000000000</max>
                    <readonly/>
                </max_memory_usage>
            </constraints>
            <allow>
                <max_memory_usage>
                    <min>6000000000</min>
                    <max>21000000000</max>
                    <readonly/>
                </max_memory_usage>
            </allow>
        </profile1>
    </profiles>
</clickhouse>
```

Reply:
> It seems it will be a bit difficult to understand, for example

Yes, I totally agree. This is why I proposed <max_in_readonly> and <min_in_readonly> as the first attempt. But it looks ugly and also not very obvious (what is minimum in readonly mode?):
```
                <max_threads>
                   <min>3</min>
                   <max>16</max>
                   <max_in_readonly>4</max_in_readonly>
                </max_threads>
```

> What will that mean?

Current behaviour (also described in docs as not the most obvious thing) is the following. Constraints and allowances are merged from multiple profiles independently. Previous constraint is not canceled by new allowance, as well as previous allowance is not canceled by new constraint. Instead profile can have one allowance and one constraint for every setting. Allowances are used in `readonly=1` mode, otherwise constraints are used.